app/src/components/fields/dynamicFieldsSection/utils.js

```diff
@@ -42,17 +45,23 @@ export const normalizeFieldsWithOptions = (fields, defaultOptionValueKey = VALUE
     if (!field?.definedValues?.length) {
       return field;
     }
+    if (!field.required && field.fieldType === OPTION_TYPE) {
+      field.definedValues.unshift({ [VALUE_NAME_KEY]: VALUE_NONE });
+    }
     const definedValues = field.definedValues.map(normalizeDefinedValue);
     let value = field.value;
-    if (!value?.length) {
+    if (!value?.length && field.fieldType !== ARRAY_TYPE) {
       value = [definedValues[0][defaultOptionValueKey]];
     }
     return { ...field, definedValues, value };
   });
 
 export const mergeFields = (savedFields, fetchedFields) =>
   fetchedFields.map((field) => {
-    const savedField = savedFields.find((item) => item.id === field.id);
+    const savedField = omit(
+      savedFields.find((item) => item.id === field.id),
+      ['definedValues'],
+    );
     return savedField ? { ...field, ...savedField } : field;
```

Comment:
According to this condition, previously `savedField` could be `undefined`, but after using the `omit` function here, it will always be an object. Please configure the use of `omit` function so that it is not applied for `savedField` with `undefined` value.